src/ForwardsTransactionIds.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace DoSomething\Gateway;
+
+trait ForwardsTransactionIds
+{
+    /**
+     * Run custom tasks before making a request.
+     *
+     * @see RestApiClient@raw
+     */
+    public function runForwardsTransactionIdsTasks($method, &$path, &$options, &$withAuthorization)
+    {
+        $transactionId = $this->getTransactionBridge()->getHeader('X-Request-ID');
+
+        // If there is no 'X-Request-ID' in the header, create one.
+        if (! $transactionId) {
+            $step = 0;
+            $transactionIdHeader = ['X-Request-ID' => microtime(true) . '-' . $step];
+        } else {
+            // Else, if there is a 'X-Request-ID' in the header, get transaction ID and increment the step at the end of Transaction ID.
+            $transactionIdBase = substr($transactionId, 0, -1);
```

Comment:
I'm a little confused on how this works here with `0, -1`. Does it start at the first item in the string at `0` and then grab the rest? Would it be able to handle a double-digit `$step`? I think this might be simpler if you just `explode()` on the `-` and then you havae the base separated from the steps in an array of items, and if you know the steps are always the second item in that array, it makes it pretty easy to work with. Might be worth a look, or maybe as a future refactor if this is all currently working fine 😄